src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -121,36 +153,66 @@ CHIP_ERROR PairingSession::DecodeMRPParametersIfPresent(TLV::Tag expectedTag, TL
     if (TLV::TagNumFromTag(tlvReader.GetTag()) == 1)
     {
         ReturnErrorOnFailure(tlvReader.Get(tlvElementValue));
-        mRemoteMRPConfig.mIdleRetransTimeout = System::Clock::Milliseconds32(tlvElementValue);
+        mRemoteSessionParam.SetMRPIdleRetransTimeout(System::Clock::Milliseconds32(tlvElementValue));
 
         // The next element is optional. If it's not present, return CHIP_NO_ERROR.
-        CHIP_ERROR err = tlvReader.Next();
-        if (err == CHIP_END_OF_TLV)
-        {
-            return tlvReader.ExitContainer(containerType);
-        }
-        ReturnErrorOnFailure(err);
+        IncrementReaderExitOnEndOfTlv(tlvReader);
     }
 
     if (TLV::TagNumFromTag(tlvReader.GetTag()) == 2)
     {
         ReturnErrorOnFailure(tlvReader.Get(tlvElementValue));
-        mRemoteMRPConfig.mActiveRetransTimeout = System::Clock::Milliseconds32(tlvElementValue);
+        mRemoteSessionParam.SetMRPActiveRetransTimeout(System::Clock::Milliseconds32(tlvElementValue));
 
         // The next element is optional. If it's not present, return CHIP_NO_ERROR.
-        CHIP_ERROR err = tlvReader.Next();
-        if (err == CHIP_END_OF_TLV)
-        {
-            return tlvReader.ExitContainer(containerType);
-        }
-        ReturnErrorOnFailure(err);
+        IncrementReaderExitOnEndOfTlv(tlvReader);
     }
 
     if (TLV::TagNumFromTag(tlvReader.GetTag()) == 3)
     {
         ReturnErrorOnFailure(tlvReader.Get(tlvElementValue));
-        mRemoteMRPConfig.mActiveThresholdTime = System::Clock::Milliseconds16(tlvElementValue);
+        mRemoteSessionParam.SetMRPActiveThresholdTime(System::Clock::Milliseconds16(tlvElementValue));
+
+        // The next element is optional. If it's not present, return CHIP_NO_ERROR.
+        IncrementReaderExitOnEndOfTlv(tlvReader);
+    }
+
+    if (TLV::TagNumFromTag(tlvReader.GetTag()) == 4)
+    {
+        ReturnErrorOnFailure(tlvReader.Get(tlvElementValue));
+        mRemoteSessionParam.SetDataModelRev(static_cast<uint16_t>(tlvElementValue));
```

Comment:
would it be worth it to use `SafeInt.h` and check `CanCastTo<uint16_t>(tlvElementValue)` to enforce that TLV underlying structure gives us values within a reaasonable range?